entity.module

```diff
@@ -86,3 +89,54 @@ function entity_module_preuninstall($module) {
     \Drupal::service('entity.bundle_plugin_installer')->uninstallBundles($entity_type, [$module]);
   }
 }
+
+/**
+ * Implements hook_query_TAG_alter().
+ *
+ * Performs access filtering for entity queries.
+ */
+function entity_query_entity_query_alter(Select $query) {
+  $entity_type_id = $query->getMetaData('entity_type');
+  if ($query->hasTag($entity_type_id . '_access')) {
+    $entity_type_manager = \Drupal::entityTypeManager();
+    $entity_type = $entity_type_manager->getDefinition($entity_type_id);
```

Comment:
_Technically_... this could be `NULL` based on the EntityTypeManagerInterface.

Reply:
Wouldn't this just throw an exception?

Reply:
Yeah, I guess that's really what we want here. I can't think of anything more appropriate.